packages/test/test-end-to-end-tests/src/test/signalPerfTelemetry.spec.ts

```diff
@@ -0,0 +1,367 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { describeCompat, itExpects } from "@fluid-private/test-version-utils";
+import { IContainer } from "@fluidframework/container-definitions/internal";
+import { ConnectionState } from "@fluidframework/container-loader";
+import { MockLogger } from "@fluidframework/telemetry-utils/internal";
+import {
+	DataObjectFactoryType,
+	ITestContainerConfig,
+	ITestFluidObject,
+	ITestObjectProvider,
+	getContainerEntryPointBackCompat,
+	timeoutPromise,
+} from "@fluidframework/test-utils/internal";
+
+const waitForSignal = async (...signallers: { once(e: "signal", l: () => void): void }[]) =>
+	Promise.all(
+		signallers.map(async (signaller, index) =>
+			timeoutPromise((resolve) => signaller.once("signal", () => resolve()), {
+				durationMs: 2000,
+				errorMsg: `Signaller[${index}] Timeout`,
+			}),
+		),
+	);
+
+describeCompat("Signal performance telemetry", "NoCompat", (getTestObjectProvider, apis) => {
+	let provider: ITestObjectProvider;
+	let dataObject: ITestFluidObject;
+	let container: IContainer;
+	let containerRuntime;
```

Comment:
untyped?

Reply:
Added a type that combines IContainerRuntimeBase with clientId and submitSignal functionality